force-app/main/default/classes/PS_CommitmentRequest.cls

```diff
@@ -194,6 +194,17 @@ public inherited sharing class PS_CommitmentRequest {
             .withBody(jsonRequestBody)
             .build();
     }
+    public static HttpRequest buildRequest(String commitmentId, String jsonRequestBody,
```

Comment:
I think this method and the method above can be combined into 1 and move the post/patch method logic out to the R`D2_CommitmentService.handleCommitment()`. Similar to the work you did in the `handleCommitmentPause()`